lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -866,13 +866,52 @@ class MongoClientInterface {
         });
     }
 
+    /**
+     * Update master with specific version if provided.
+     * @param {Object} c bucket collection
+     * @param {String} bucketName bucket name
+     * @param {String} objName object name
+     * @param {Object} objVal object metadata
+     * @param {Object} params params
+     * @param {String} params.masterVersionId master version id
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
+    putObjectMasterVer(c, bucketName, objName, objVal, params, log, cb) {
+        const { masterVersionId, vFormat } = params;
+        if (masterVersionId) {
+            // eslint-disable-next-line no-param-reassign
+            objVal.versionId = masterVersionId;
+        }
+        const masterKey = formatMasterKey(objName, vFormat);
+        c.update({
+            _id: masterKey,
```

Comment:
Can you also update the microVersionId ? (look at object model)

`models/ObjectMD.js:    updateMicroVersionId()`

This will be used to know internally that there was an update.